src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -284,6 +284,22 @@
         "DELTA",  # https://github.com/delta-io/delta
         "XML",  # https://github.com/databricks/spark-xml
     ),
+    # Adding Hint related segments so they are not treated as generic comments
+    # https://spark.apache.org/docs/latest/sql-ref-syntax-qry-select-hints.html
+    StartHintCommentSegment=StringParser(
+        "/*+", KeywordSegment, name="start_hint_comment"
+    ),
+    EndHintCommentSegment=StringParser("*/", KeywordSegment, name="end_hint_comment"),
+    HintCommentGrammar=Sequence(
+        Sequence(
+            Ref("StartHintCommentSegment"),
+            Delimited(
+                Ref("FunctionSegment"),
+                min_delimiters=1,
+                terminator=Ref("EndHintCommentSegment"),
+            ),
+        ),
+    ),
```

Comment:
I'd make this a Segment and give it a type.

I also think you don't want `min_delimters=1` as some of your examples have no delimiters.

And some of your Function calls have no brackets.

So I'd suggest two new segments like this instead:

```py
@spark3_dialect.segment()
class HintSegment(BaseSegment):
    """A hint function."""

    type = "function"
    match_grammar = Sequence(
        Sequence(
            Ref("FunctionNameSegment"),
            Bracketed(
                Ref(
                    "FunctionContentsGrammar",
                    # The brackets might be empty for some functions...
                    optional=True,
                    ephemeral_name="FunctionContentsGrammar",
                ),
                optional=True,
            ),
        ),
    )

@spark3_dialect.segment()
class HintCommentSegment(BaseSegment):
    """Spark Select Hints.

    https://spark.apache.org/docs/latest/sql-ref-syntax-qry-select-hints.html
    """

    type = "hint"
    match_grammar = Sequence(
        Sequence(
            Ref("StartHintCommentSegment"),
            Delimited(
                Ref("HintSegment"),
                terminator=Ref("EndHintCommentSegment"),
            ),
            Ref("EndHintCommentSegment"),
        ),
    )
```